jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -173,6 +174,21 @@ static JarType determineJarType(Path jarPath) throws IOException {
     }
   }
 
+  private static void addDependency(
+      FileEntriesLayer.Builder layerbuilder,
+      Path depPath,
+      Path jarParent,
+      AbsoluteUnixPath pathOnContainer) {
+    File depFile = new File(jarParent.resolve(depPath).toString());
+    if (!depFile.exists()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Dependency: %s needs to be in the same parent directory as the JAR.",
+              depPath.toString()));
```

Comment:
I think this will cause confusion when a dependency is nested. It reads like such a dependency should be in the same parent directory. How about something like

"Dependency required by the JAR (as specified in 'Class-Path:' in the JAR manifest) doesn't exist: <full path of jarParent.resolve(depPath)>"

? And then we can also pass the full path instead of two argument (`depPath` and `jarParent`) to the method. WDYT?

Reply:
Oh yeah, that can be really confusing. Passing in the full dependency path is a good idea.